### PR TITLE
Up 4880 unit test layoutpackage

### PR DESCRIPTION
**Pull request by stalele**
##### Checklist

-   [X] the [individual contributor license agreement][] is signed
-   [X] commit message follows [commit guidelines][]
-   [X] tests are included
-   [ ] documentation is changed or added
-   [ ] [message properties][] have been updated with new phrases
-   [ ] view conforms with [WCAG 2.0 AA][]

##### Description of change

Added unit test cases for  org.apereo.portal.rest.layout package 
Jira-Ticket -> https://issues.jasig.org/browse/UP-4880

[individual contributor license agreement]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#individual-contributor-license-agreement
[commit guidelines]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#commit
[message properties]: https://github.com/Jasig/uPortal/tree/master/uportal-war/src/main/resources/properties/i18n
[WCAG 2.0 AA]: https://www.w3.org/WAI/WCAG20/quickref/?levels=aaa&technologies=smil%2Cpdf%2Cflash%2Csl

### CODE REVIEW

**Comment by ChristianMurphy** [comment]
:notebook: Some commits overlap with #935
Commits including and after ba19f40 are unique to this PR.

File: uPortal-api/uPortal-api-rest/src/test/java/org/apereo/portal/rest/layout/TabListOfNodesTest.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+package org.apereo.portal.rest.layout;
+
+
+import java.io.StringReader;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+
+import junit.framework.Assert;
```
Comment (ChristianMurphy) [inline uPortal-api/uPortal-api-rest/src/test/java/org/apereo/portal/rest/layout/TabListOfNodesTest.java]: `junit.framework.Assert` is deprecated, prefer `org.junit.Assert` instead. 

ref: https://stackoverflow.com/questions/24013210/assert-in-junit-framework-has-been-deprecated-what-use-next

File: uPortal-api/uPortal-api-rest/src/test/java/org/apereo/portal/rest/PagsRESTControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+ * <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apereo.portal.rest;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import junit.framework.Assert;
```
Comment (ChristianMurphy) [inline uPortal-api/uPortal-api-rest/src/test/java/org/apereo/portal/rest/PagsRESTControllerTest.java]: `junit.framework.Assert` is deprecated, prefer `org.junit.Assert` instead. 

ref: https://stackoverflow.com/questions/24013210/assert-in-junit-framework-has-been-deprecated-what-use-next

File: uPortal-api/uPortal-api-rest/src/test/java/org/apereo/portal/rest/layout/TabListOfNodesTest.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apereo.portal.rest.layout;
+
+import java.io.StringReader;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import junit.framework.Assert;
```
Comment (ChristianMurphy) [inline uPortal-api/uPortal-api-rest/src/test/java/org/apereo/portal/rest/layout/TabListOfNodesTest.java]: `junit.framework.Assert` is deprecated, prefer `org.junit.Assert` instead. 

ref: https://stackoverflow.com/questions/24013210/assert-in-junit-framework-has-been-deprecated-what-use-next

**Comment by ChristianMurphy** [comment]
Linux build is running into unrelated out of memory exception.
3ccbe5b4c89b2365614ebfd26e255166aceccf30 from #937 will resolve.

File: uPortal-api/uPortal-api-rest/src/test/java/org/apereo/portal/rest/layout/LayoutJsonV1RenderingControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+public class LayoutJsonV1RenderingControllerTest {
+    @InjectMocks LayoutJsonV43RenderingController controller;
+    @Mock private HttpServletRequest req;
+
+    @Mock private HttpServletResponse res;
+
+    @Mock private IPortalRenderingPipeline portalRenderingPipeline;
```
Comment (bjagg) [inline uPortal-api/uPortal-api-rest/src/test/java/org/apereo/portal/rest/layout/LayoutJsonV1RenderingControllerTest.java]: Why the blank likes around HttpServletResponse var?

File: uPortal-api/uPortal-api-rest/src/test/java/org/apereo/portal/rest/layout/LayoutJsonV43RenderingControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+import org.mockito.MockitoAnnotations;
+
+public class LayoutJsonV43RenderingControllerTest {
+
+    @InjectMocks LayoutJsonV43RenderingController controller;
+    @Mock private HttpServletRequest req;
+
+    @Mock private HttpServletResponse res;
+
+    @Mock private IPortalRenderingPipeline portalRenderingPipeline;
```
Comment (bjagg) [inline uPortal-api/uPortal-api-rest/src/test/java/org/apereo/portal/rest/layout/LayoutJsonV43RenderingControllerTest.java]: Same as above

**Comment by ChristianMurphy** [comment]
Changes merged as part of https://github.com/Jasig/uPortal/pull/940

**Pull request closed without merging**